collect_app/src/main/java/org/odk/collect/android/listeners/SelectOneItemClickListener.java

```diff
@@ -0,0 +1,5 @@
+package org.odk.collect.android.listeners;
+
+public interface SelectOneItemClickListener extends SelectItemClickListener {
+    void onClearNextLevelsOfCascadingSelect();
```

Comment:
I think this is broken either way even on master and w didn't have any test for that clearing. Can I file a separate issue for that to fix it and add handle in a better way and here just remove?

Reply:
Unfortunately, I don't think we should just remove it because it does address a real issue and works well in the common case. Is it not possible to move the behavior just into the `onItemClicked` implementation for `ItemsetWidget`?

Definitely good to file an issue to review it and add tests, though. 👍

Reply:
Ok I made a mistake (clearing works as before) I used the form from the doc https://docs.google.com/spreadsheets/d/1CCjRRHCyJXaSEBHPjMWrGotnORR4BI49PoON6qK01BE/edit#gid=1 which seems broken because it doesn't display the last level. It's called new cascading select so it indicates that something has changed do you know anything about it? I know that the old version  has two columns (for filters) and I was able to transform it and now it works but I'm curious what's going on.
Regarding the listener you are right I don't need that second listener. I already removed it.

Reply:
That form is just crazy. I have no idea why it was like that and I just merged https://github.com/XLSForm/xlsform.github.io/pull/196 to improve it.

I don't think `onClearNextLevelsOfCascadingSelect` would be relevant to that form, though. It's a special case only for "external itemsets" (http://xlsform.org/en/#external-selects).

Reply:
Yea I agree with internal choices it works without that fix but now I can confirm that it also works with external choices. At least as it worked before.

Reply:
Great. Just be careful when saying "external choices" because that can mean three things: `search()`, "external itemsets" (`ItemsetWidget`) or external secondary instances.